index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
> Why do you want [Maybe] to be a Semigroup/Monoid?

To enable `filter` to be defined in terms of `chain`, `empty`, and `of` as suggested by @joneshf in https://github.com/plaid/sanctuary/pull/10#issuecomment-74254613. `filter` needn't be defined this way, of course, but I love the idea of deriving it in the same manner for all such types.